### PR TITLE
Add ocaml-dune/binary-distribution

### PR DESCRIPTION
**Pull request by mtelvers**

### CODE REVIEW

File: src/cluster.ml [hunk 1]
```diff
@@ -68,6 +69,7 @@ type service = [
   | `Ocamlorg_images of string               (* Base Image builder @ images.ci.ocaml.org *)
   | `OCamlorg_v3b of string                  (* OCaml website @ v3b.ocaml.org aka www.ocaml.org *)
   | `OCamlorg_v3c of string                  (* Staging OCaml website @ staging.ocaml.org *)
+  | `Dune_distribution_docker of string      (* Dune binary distribution website *)
```
Comment (maiste) [inline src/cluster.ml:72]:
```suggestion
  | `Dune_binary_distribution      (* Dune binary distribution website *)
```
We can stick to for now. If the infrastructure gets more complex, we can change the naming at this point/.

File: src/cluster.ml [hunk 2]
```diff
@@ -182,6 +184,7 @@ let pull_and_serve multi_hash = function
   | `Ocamlorg_images name -> pull_and_serve (module Ocamlorg_images) ~name `Service multi_hash
   | `OCamlorg_v3b name -> pull_and_serve (module V3b_docker) ~name `Service multi_hash
   | `OCamlorg_v3c name -> pull_and_serve (module V3c_docker) ~name `Service multi_hash
+  | `Dune_distribution_docker name -> pull_and_serve (module Dune_distribution_docker) ~name `Service multi_hash
```
Comment (maiste) [inline src/cluster.ml:187]:
```suggestion
  | `Dune_binary_distribution -> pull_and_serve (module Dune_binary_distribution) ~name `Service multi_hash
```

File: src/cluster.ml [hunk 3]
```diff
@@ -41,6 +41,7 @@ module Ocamlorg_images = Current_docker.Make(struct let docker_context = Some "c
 module Docker_aws = Current_docker.Make(struct let docker_context = Some "awsecs" end)
 module V3b_docker = Current_docker.Make(struct let docker_context = Some "v3b.ocaml.org" end)
 module V3c_docker = Current_docker.Make(struct let docker_context = Some "v3c.ocaml.org" end)
+module Dune_distribution_docker = Current_docker.Make(struct let docker_context = Some "download.dune.ci.dev" end)
```
Comment (maiste) [inline src/cluster.ml:44]:
```suggestion
module Dune_binary_distribution = Current_docker.Make(struct let docker_context = Some "preview.dune.build" end)
```

File: create-config.sh [hunk 4]
```diff
@@ -18,6 +18,7 @@ docker context create "opam.ci.ocaml.org" --description "OCaml - Opam Repo CI" -
 docker context create "ocaml.ci.dev" --description "Tarides - OCaml CI" --docker "host=ssh://root@ocaml.ci.dev"
 docker context create "check.ci.ocaml.org" --description "OCaml - Opam Health Check" --docker "host=ssh://root@check.ci.ocaml.org"
 docker context create "registry.ci.dev" --description "Docker Registry" --docker "host=ssh://root@registry.ci.dev"
+docker context create "download.dune.ci.dev" --description "Dune Binary Distribution" --docker "host=ssh://root@download.dune.ci.dev"
```
Comment (maiste) [inline create-config.sh:21]:
```suggestion
docker context create "preview.dune.build" --description "Dune Binary Distribution" --docker "host=ssh://root@preview.dune.build"
```

**Review by maiste — COMMENTED** [review]
I changed the target to point to the preview endpoint.

File: create-config.sh (hunk 4 above)
Comment (mtelvers) [reply create-config.sh]: I think we should use get.dune.build here to avoid the need to change it in the future.

File: src/cluster.ml (hunk 3 above)
Comment (mtelvers) [reply src/cluster.ml]: I think we should use get.dune.build

File: src/cluster.ml (hunk 2 above)
Comment (mtelvers) [reply src/cluster.ml]: We should stick with the _docker naming convention here.

File: src/cluster.ml (hunk 1 above)
Comment (mtelvers) [reply src/cluster.ml]: Yes, I've updated this.  It still needs to be `of string`, though.

**Review by shonfeder — APPROVED** [review]

**Comment by shonfeder** [comment]
I've pushed the updated documentation build to clear the linting error.

This has already been deployed to live, and I have pending work that will need to resolve merge conflicts with this change, so I'm gonna merge it in to master.

**Pull request merged**